### PR TITLE
Fix divide zero issue in mat4.toSRT which causes node._lrot with NaN values.

### PR DESCRIPTION
**Pull request by dumganhar**
Re: #16929

If node._lrot.x is NaN, it will be serialized to scene data with `null` value.

企业微信截图_b5056fb5-4838-440b-9b90-774731779425

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/core/math/mat4.ts [hunk 1]
```diff
@@ -930,27 +900,43 @@ export class Mat4 extends ValueType {
...
-            Vec3.set(v, m.m12, m.m13, m.m14);
-        }
-        if (q) {
+
+        if (sx !== 0) {
             m3_1.m00 = m.m00 / sx;
             m3_1.m01 = m.m01 / sx;
             m3_1.m02 = m.m02 / sx;
+        } else {
+            m3_1.m00 = m3_1.m01 = m3_1.m02 = 0;
```
Comment (minggo) [inline cocos/core/math/mat4.ts]: Is else needed? If length is 0, then all values should be 0.

File: cocos/core/math/mat4.ts [hunk 2]
```diff
@@ -930,27 +900,43 @@ export class Mat4 extends ValueType {
...
+
+        const det = Mat3.determinant(m3_1);
+        if (s) {
+            s.x = sx;
+            s.y = sy;
+            s.z = sz;
+            if (det < 0) {
+                s.x *= -1;
+            }
+        }
```
Comment (star-e) [inline cocos/core/math/mat4.ts:940]: Here I don't understand, why inverse s.x?

Comment (dumganhar) [reply cocos/core/math/mat4.ts:940]: https://github.com/cocos/cocos-engine/blob/v3.8.3/cocos/core/math/mat4.ts#L904

It's from the original logic。

### There is an explain from AI:


When the determinant is less than 0, it signifies two important physical meanings:

1. Reversal of Orientation: The sign of the determinant indicates whether a transformation preserves the orientation of space. When the determinant is negative, it means that the transformation reverses the orientation of space, flipping from the positive direction to the negative direction. Geometrically, this implies that objects are oriented oppositely after the transformation compared to the original orientation.
2. Volume Reversal: The absolute value of the determinant represents the scaling factor of the space volume defined by the transformation. When the determinant is negative, it implies that the transformation results in a reversal of the space volume, meaning that after the transformation, the volume of objects in space is reversed, becoming opposite.

Comment (dumganhar) [reply cocos/core/math/mat4.ts:940]: @star-e I found some reference in Ogre engine:

https://github.com/OGRECave/ogre/blob/26c02c910dcc3c90580111403e6e40285ee785ad/OgreMain/src/OgreMatrix3.cpp#L626

**Review by minggo — APPROVED** [review]

File: cocos/core/math/mat4.ts [hunk 3]
```diff
@@ -935,22 +911,36 @@ export class Mat4 extends ValueType {
...
+
+        // Scale too close to zero, can't decompose rotation.
+        if (sx === 0 || sy === 0 || sz === 0) {
+            if (q) {
+                Quat.identity(q);
+            }
+            return;
         }
+
+        m3_1.m00 = m.m00 / sx;
```
Comment (dumganhar) [inline cocos/core/math/mat4.ts:923]: The original code did not check whether the sx value is 0, and divide 0 will make m3_1.m00 be a NaN value.

**Review by GengineJS — APPROVED** [review]

File: cocos/core/math/mat4.ts (hunk 2 above)
Comment (star-e) [reply cocos/core/math/mat4.ts:940]: I think it mirrors the x-axis. As long as the conversion is consistent, it will be fine.

**Review by star-e — APPROVED** [review]

**Pull request merged**